### PR TITLE
Couple of fixes

### PR DESCRIPTION
**Pull request by dreamfall**
- date validation backend fix
- reset email password fix for admins
- add award to the form if user answered in eligibility that he had an award (only for trade)

### CODE REVIEW

File: app/controllers/form_controller.rb [hunk 1]
```diff
@@ -108,7 +108,22 @@ def edit_form
       end
 
       if @form_answer.document["queen_award_holder"].blank?
-        @form_answer.document = @form_answer.document.merge(queen_award_holder: queen_award_holder)
+        holder = if @form_answer.trade?
+          @form_answer.trade_eligibility.current_holder_of_qae_for_trade? ? "yes" : "no"
+        else
+          queen_award_holder
+        end
```
Comment (matthewford) [inline app/controllers/form_controller.rb]: `end` at 115, 8 is not aligned with `if` at 111, 17

File: app/controllers/form_controller.rb [hunk 2]
```diff
@@ -108,7 +108,22 @@ def edit_form
...
+
+        @form_answer.document = @form_answer.document.merge(queen_award_holder: holder)
+
+        if holder == "yes" && @form_answer.trade?
+          year = @form_answer.trade_eligibility.qae_for_trade_award_year
+          if year.to_i < AwardYear.current.year - 5 || !@form_answer.trade_eligibility.current_holder_of_qae_for_trade?
```
Comment (matthewford) [inline app/controllers/form_controller.rb]: Line is too long. [119/100]

File: app/controllers/form_controller.rb [hunk 3]
```diff
@@ -108,7 +108,22 @@ def edit_form
...
+            @form_answer.document = @form_answer.document.merge(queen_award_holder: "no")
+          else
+            @form_answer.document = @form_answer.document.merge(queen_award_holder_details: [{category: "international_trade", year: year.to_s}.to_json].to_json)
```
Comment (matthewford) [inline app/controllers/form_controller.rb]: Line is too long. [161/100]<br>Space inside { missing.<br>Space inside } missing.

File: app/controllers/form_controller.rb [hunk 4]
```diff
@@ -108,7 +108,22 @@ def edit_form
       end
 
       if @form_answer.document["queen_award_holder"].blank?
-        @form_answer.document = @form_answer.document.merge(queen_award_holder: queen_award_holder)
+        holder = if @form_answer.trade?
+          @form_answer.trade_eligibility.current_holder_of_qae_for_trade? ? "yes" : "no"
+        else
```
Comment (matthewford) [inline app/controllers/form_controller.rb]: Align `else` with `if`.

File: app/controllers/form_controller.rb [hunk 5]
```diff
@@ -108,7 +108,22 @@ def edit_form
       end
 
       if @form_answer.document["queen_award_holder"].blank?
-        @form_answer.document = @form_answer.document.merge(queen_award_holder: queen_award_holder)
+        holder = if @form_answer.trade?
+          @form_answer.trade_eligibility.current_holder_of_qae_for_trade? ? "yes" : "no"
```
Comment (matthewford) [inline app/controllers/form_controller.rb]: Use 2 (not -7) spaces for indentation.

File: spec/controllers/form_controller_spec.rb [hunk 6]
```diff
@@ -38,4 +39,74 @@
...
+          any_dips_over_the_last_three_years: true,
+          current_holder_of_qae_for_trade: true,
+          qae_for_trade_award_year: "2015"
+        )
+      end
+
+      it "adds award if it is 5 or less years old" do
+        get :edit_form, id: form_answer.id
+
+        expect(form_answer.reload.document["queen_award_holder_details"]).to eq([{category: "international_trade", year: "2015"}.to_json].to_json)
```
Comment (matthewford) [inline spec/controllers/form_controller_spec.rb]: Line is too long. [146/100]<br>Space inside { missing.<br>Space inside } missing.

File: spec/controllers/form_controller_spec.rb [hunk 7]
```diff
@@ -38,4 +39,74 @@
...
+        expect(form_answer.document["queen_award_holder"]).to eq("yes")
+      end
+
+      it "ingonres award if it is 5 or more years old" do
+        trade_el = form_answer.trade_eligibility
+        trade_el.update!({
```
Comment (matthewford) [inline spec/controllers/form_controller_spec.rb]: Redundant curly braces around a hash parameter.

File: spec/controllers/form_controller_spec.rb [hunk 8]
```diff
@@ -38,4 +39,74 @@
...
+        trade_el.update!(
+          sales_above_100_000_pounds: "yes",
+          any_dips_over_the_last_three_years: true,
+          current_holder_of_qae_for_trade: false
+        )
+        get :edit_form, id: form_answer.id
+
+        expect(form_answer.reload.document["queen_award_holder_details"]).to be_nil
+        expect(form_answer.document["queen_award_holder"]).to eq("no")
+
```
Comment (matthewford) [inline spec/controllers/form_controller_spec.rb]: Extra empty line detected at block body end.

File: spec/controllers/form_controller_spec.rb [hunk 9]
```diff
@@ -38,4 +39,74 @@
...
+        get :edit_form, id: form_answer.id
+
+        expect(form_answer.reload.document["queen_award_holder_details"]).to eq([{category: "international_trade", year: "2015"}.to_json].to_json)
+        expect(form_answer.document["queen_award_holder"]).to eq("yes")
+      end
+
+      it "ingonres award if it is 5 or more years old" do
+        trade_el = form_answer.trade_eligibility
+        trade_el.update!({
+          sales_above_100_000_pounds: "yes",
```
Comment (matthewford) [inline spec/controllers/form_controller_spec.rb]: Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.

File: spec/controllers/form_controller_spec.rb [hunk 10]
```diff
@@ -38,4 +39,74 @@
...
+          any_dips_over_the_last_three_years: true,
+          current_holder_of_qae_for_trade: true,
+          qae_for_trade_award_year: "before_2011"
+        })
```
Comment (matthewford) [inline spec/controllers/form_controller_spec.rb]: Indent the right brace the same as the first position after the preceding left parenthesis.

File: app/controllers/form_controller.rb [hunk 11]
```diff
@@ -108,7 +108,24 @@ def edit_form
       end
 
       if @form_answer.document["queen_award_holder"].blank?
-        @form_answer.document = @form_answer.document.merge(queen_award_holder: queen_award_holder)
+        eligibility_holder = @form_answer.trade_eligibility.current_holder_of_qae_for_trade?
+        holder = if @form_answer.trade?
+          eligibility_holder ? "yes" : "no"
+        else
+          queen_award_holder
+        end
```
Comment (matthewford) [inline app/controllers/form_controller.rb]: `end` at 116, 8 is not aligned with `if` at 112, 17

File: app/controllers/form_controller.rb [hunk 12]
```diff
@@ -108,7 +108,24 @@ def edit_form
       end
 
       if @form_answer.document["queen_award_holder"].blank?
-        @form_answer.document = @form_answer.document.merge(queen_award_holder: queen_award_holder)
+        eligibility_holder = @form_answer.trade_eligibility.current_holder_of_qae_for_trade?
+        holder = if @form_answer.trade?
+          eligibility_holder ? "yes" : "no"
+        else
```
Comment (matthewford) [inline app/controllers/form_controller.rb]: Align `else` with `if`.

File: app/controllers/form_controller.rb [hunk 13]
```diff
@@ -108,7 +108,24 @@ def edit_form
       end
 
       if @form_answer.document["queen_award_holder"].blank?
-        @form_answer.document = @form_answer.document.merge(queen_award_holder: queen_award_holder)
+        eligibility_holder = @form_answer.trade_eligibility.current_holder_of_qae_for_trade?
+        holder = if @form_answer.trade?
+          eligibility_holder ? "yes" : "no"
```
Comment (matthewford) [inline app/controllers/form_controller.rb]: Use 2 (not -7) spaces for indentation.

**Comment by RuslanHamidullin** [comment]
super :+1:

**Comment by RuslanHamidullin** [comment]
waiting for travis

**Pull request merged**